src/PrestaShopBundle/Form/Admin/ShopParameters/CustomerPreferences/CustomerPreferencesFormHandler.php

```diff
@@ -0,0 +1,158 @@
+<?php
+/**
+ * 2007-2018 PrestaShop
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2018 PrestaShop SA
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShopBundle\Form\Admin\ShopParameters\CustomerPreferences;
+
+use PrestaShop\PrestaShop\Core\Form\FormDataProviderInterface;
+use PrestaShop\PrestaShop\Core\Form\FormHandlerInterface;
+use PrestaShopBundle\Entity\Repository\TabRepository;
+use PrestaShopBundle\Service\Hook\HookDispatcher;
+use Symfony\Component\Form\FormFactoryInterface;
+use Symfony\Component\Translation\TranslatorInterface;
+
+/**
+ * Class manages "Configure > Shop Parameters > Customer Settings" page
+ * form handling.
+ */
+class CustomerPreferencesFormHandler implements FormHandlerInterface
+{
+    /**
+     * @var FormFactoryInterface
+     */
+    private $formFactory;
+
+    /**
+     * @var FormDataProviderInterface
+     */
+    private $dataProvider;
+
+    /**
+     * @var TranslatorInterface
+     */
+    private $translator;
+
+    /**
+     * @var HookDispatcher
+     */
+    private $dispatcher;
+
+    /**
+     * @var TabRepository
+     */
+    private $tabRepository;
+
+    public function __construct(
+        FormFactoryInterface $formFactory,
+        FormDataProviderInterface $dataProvider,
+        TranslatorInterface $translator,
+        HookDispatcher $dispatcher,
+        TabRepository $tabRepository
```

Comment:
maybe create TabRepositoryInterface in Core namespace and depend on it?

Reply:
I have an "deja vu" feeling... aren't you working on a Tab repository already @rokas?

Reply:
Don't we have already a Repository Interface in Core?

Reply:
Couldn't find any.